packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.base.tsx

```diff
@@ -77,7 +77,11 @@ function useWeeks(
     }
 
     return returnValue;
-  }, [props]);
+    // TODO: check with @MLoughry -- this was missing deps on getSetRefCallback and onSelectDate.
+    // It was easy to prevent getSetRefCallback from changing every render, but onSelectDate is
+    // harder (and likely props will change a lot too). Should we ignore the violation, add a dep
+    // on onSelectDate, or fix it "properly"?
+  }, [props, getSetRefCallback]);
```

Comment:
We should add `onSelectDate` here, and probably make `onSelectDate` a memo'd callback as well.

Reply:
I tried making `onSelectDate` memo'd, but that turns out to require a huge cascade of changes throughout Calendar as well. That seemed likely to be out of scope for a change like this, so I just disabled the rule for this line instead.

I also noticed while looking through the calendar code that some of the hooks (particularly `useDateState`) call state setters while rendering, which seems unsafe, particularly for concurrent mode.

Reply:
@ecraig12345 Regarding calling state setters during render: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops

> ... you can update the state right during rendering. React will re-run the component with updated state immediately after exiting the first render so it wouldn’t be expensive.

Reply:
Oh interesting, that is not what I would have expected from experience with class components.